src/Compilers/CSharp/Portable/Symbols/TypeParameterSymbol.cs

```diff
@@ -369,69 +358,76 @@ internal ImmutableArray<NamedTypeSymbol> AllEffectiveInterfacesWithDefinitionUse
         /// to allow derived classes to ensure constraints within the containing
         /// type or method are resolved in a consistent order, regardless of the
         /// order the callers query individual type parameters.
-        /// The `early` parameter indicates whether the constraints can be from
-        /// the initial phase of constraint checking where the constraint types may
-        /// still contain invalid or duplicate types.
         /// </summary>
-        internal abstract void EnsureAllConstraintsAreResolved(bool early);
+        internal abstract void EnsureAllConstraintsAreResolved();
 
         /// <summary>
         /// Helper method to force type parameter constraints to be resolved.
         /// </summary>
-        protected static void EnsureAllConstraintsAreResolved(ImmutableArray<TypeParameterSymbol> typeParameters, bool early)
+        protected static void EnsureAllConstraintsAreResolved(ImmutableArray<TypeParameterSymbol> typeParameters)
         {
             foreach (var typeParameter in typeParameters)
             {
                 // Invoke any method that forces constraints to be resolved.
-                var unused = typeParameter.GetConstraintTypes(ConsList<TypeParameterSymbol>.Empty, early);
+                var unused = typeParameter.GetConstraintTypes(ConsList<TypeParameterSymbol>.Empty);
             }
         }
 
-        internal abstract ImmutableArray<TypeWithAnnotations> GetConstraintTypes(ConsList<TypeParameterSymbol> inProgress, bool early);
+        internal abstract ImmutableArray<TypeWithAnnotations> GetConstraintTypes(ConsList<TypeParameterSymbol> inProgress);
 
         internal abstract ImmutableArray<NamedTypeSymbol> GetInterfaces(ConsList<TypeParameterSymbol> inProgress);
 
         internal abstract NamedTypeSymbol GetEffectiveBaseClass(ConsList<TypeParameterSymbol> inProgress);
 
         internal abstract TypeSymbol GetDeducedBaseType(ConsList<TypeParameterSymbol> inProgress);
 
+        private static bool ConstraintImpliesReferenceType(TypeSymbol constraint)
```

Comment:
>ConstraintImpliesReferenceType [](start = 28, length = 30)

nit: I'd be tempted to name this method like the one below, but singular `IsReferenceTypeFromConstraintType`. I was thrown off by the distinction between "is" and "implies".
Along those lines, this method could be a local function inside of `IsReferenceTypeFromConstraintTypes`.

Reply:
This method and others in this file are simply restored to their original state prior to [https://github.com/dotnet/roslyn/pull/27525](https://github.com/dotnet/roslyn/pull/27525). 

---
In reply to: [280582299](https://github.com/dotnet/roslyn/pull/35375#discussion_r280582299) [](ancestors = 280582299)

Reply:
This include names, etc.

---
In reply to: [280583028](https://github.com/dotnet/roslyn/pull/35375#discussion_r280583028) [](ancestors = 280583028,280582299)